code/modules/clothing/suits/marine_armor.dm

```diff
@@ -288,51 +288,265 @@ var/list/squad_colors = list(rgb(230,25,25), rgb(255,195,45), rgb(200,100,200),
+	var/B18_tox_cooldown = null
+	var/B18_pain_cooldown = null
+	var/B18_automed_on = TRUE
+	var/B18_automed_damage = 50
+	var/B18_automed_pain = 60
+	var/obj/item/device/healthanalyzer/B18_analyzer = null
+	supporting_limbs = list(UPPER_TORSO, LOWER_TORSO, ARMS, LEGS, FEET) //B18 effectively auto-splints these.
 	unacidable = 1
 	req_access = list(ACCESS_MARINE_SPECPREP)
 
 	New(loc,expected_type 	= type,
 		new_name[] 		= list(MAP_ICE_COLONY = "\improper B18 defensive snow armor"))
 		..(loc,expected_type,new_name)
 
+/obj/item/clothing/suit/storage/marine/specialist/New()
+	. = ..()
+	B18_analyzer = new /obj/item/device/healthanalyzer
+
+/obj/item/clothing/suit/storage/marine/specialist/Dispose()
+	wearer = null
+	b18automed_turn_off()
+	cdel(B18_analyzer)
+	. = ..()
+
+/obj/item/clothing/suit/storage/marine/specialist/dropped(mob/user)
+	. = ..()
+	wearer = null
+	b18automed_turn_off()
+
+/obj/item/clothing/suit/storage/marine/specialist/equipped(mob/living/carbon/human/user, slot)
+	. = ..()
+	if(slot == WEAR_JACKET)
+		wearer = user
+
 /obj/item/clothing/suit/storage/marine/specialist/mob_can_equip(mob/M, slot, disable_warning = 0)
 	. = ..()
 	if(.)
 		if(!allowed(M))
 			to_chat(M, "<span class='warning'>[src] flashes a warning sign indicating unauthorized use!</span>")
 			return 0
 
-/obj/item/clothing/suit/storage/marine/specialist/verb/inject()
-	set name = "Create Injector"
-	set category = "Object"
-	set src in usr
-
-	if(!usr.canmove || usr.stat || usr.is_mob_restrained())
-		return 0
+/obj/item/clothing/suit/storage/marine/specialist/proc/b18automed_turn_off(mob/living/carbon/human/user)
+	B18_automed_on = FALSE
+	processing_objects.Remove(src)
+	to_chat(user, "<span class='warning'>[src] lets out a beep as its automedical suite deactivates.</span>")
+	playsound(src,'sound/machines/click.ogg', 15, 0, 1)
+
+/obj/item/clothing/suit/storage/marine/specialist/proc/b18automed_turn_on(mob/living/carbon/human/user)
+	B18_automed_on = TRUE
+	processing_objects.Add(src)
+	to_chat(user, "<span class='notice'>[src] lets out a hum as its automedical suite activates.</span>")
+	playsound(src,'sound/mecha/nominal.ogg', 15, 0, 1)
+
+/obj/item/clothing/suit/storage/marine/specialist/process()
+	if(!B18_automed_on)
+		processing_objects.Remove(src)
+		return
+	if(!wearer)
+		processing_objects.Remove(src)
+		return
 
-	if(!injections)
-		to_chat(usr, "Your armor is all out of injectors.")
-		return 0
+	var/list/details = list()
+	var/dose_administered = null
+	var/tricordrazine = REAGENTS_OVERDOSE - (wearer.reagents.get_reagent_amount("tricordrazine") + 0.5)
+
+	if(wearer.getFireLoss() > B18_automed_damage && !B18_burn_cooldown)
+		var/dermaline = REAGENTS_OVERDOSE*0.5 - (wearer.reagents.get_reagent_amount("dermaline") + 0.5)
+		var/kelotane = REAGENTS_OVERDOSE - (wearer.reagents.get_reagent_amount("kelotane") + 0.5)
+		if(dermaline < REAGENTS_OVERDOSE*0.5)
+			wearer.reagents.add_reagent("dermaline",REAGENTS_OVERDOSE*0.5 - dermaline + 0.5)
```

Comment:
This can horribly end up with the user being dosed with exponential amount of the drug during OD.
If the mob has 40 u of dermaline, the var will result - 24.5, and when summed during add_reagent(), the mob will be end up being pumped in another 40 units due to minus minus.
Clamp this, also halve the dose amounts (tweak the cool downs if necessary), else the mob will be prone to OD from medical crew fuck ups, or a single impulsive use of an autoinjector amongst other reasons.